classes/user/Repository.php

```diff
@@ -330,12 +330,7 @@ public function mergeUsers(int $oldUserId, int $newUserId)
             Repo::submissionFile()->edit($submissionFile, ['uploaderUserId' => $newUserId]);
         }
 
-        $noteDao = DAORegistry::getDAO('NoteDAO'); /** @var NoteDAO $noteDao */
-        $notes = $noteDao->getByUserId($oldUserId);
-        while ($note = $notes->next()) {
-            $note->setUserId($newUserId);
-            $noteDao->updateObject($note);
-        }
+        Note::withUserId($oldUserId)->each()->update(['userId' => $newUserId]);
```

Comment:
This will materialize each `Note` object individually, which can be an expensive operation. I'd suggest adding a function to the `Repository` class to handle this case -- see `lib/pkp/classes/notification/Repository.php` for an example in the `transfer()` function. (It would be possible to remove the `each()` to do it here -- but that would require us to add `snake_case` column names outside of the model code, which we'd like to avoid.)